admin/class-yoast-notification-center.php

```diff
@@ -130,50 +221,333 @@ public function set_transient() {
 	 * @param Yoast_Notification $notification Notification object instance.
 	 */
 	public function add_notification( Yoast_Notification $notification ) {
+		// Don't add persistent notifications that have already been added.
+		$notification_id = $notification->get_id();
+		if ( ! empty( $notification_id ) ) {
+			$found = $this->get_notification_by_id( $notification->get_id() );
+			if ( ! is_null( $found ) ) {
+				return;
+			}
+		}
+
+		// Add to list.
 		$this->notifications[] = $notification;
 	}
 
 	/**
+	 * Get the notification by ID
+	 *
+	 * @param string $notification_id The ID of the notification to search for.
+	 *
+	 * @return null|Yoast_Notification
+	 */
+	public function get_notification_by_id( $notification_id ) {
+		foreach ( $this->notifications as $notification ) {
+			if ( $notification_id === $notification->get_id() ) {
+				return $notification;
+			}
+		}
+
+		return null;
+	}
+
+	/**
 	 * Display the notifications
 	 */
 	public function display_notifications() {
 
-		$ids = array();
+		// Never display notifications for network admin.
+		if ( function_exists( 'is_network_admin' ) && is_network_admin() ) {
+			return;
+		}
+
+		$sorted_notifications = $this->get_sorted_notifications();
+		foreach ( $sorted_notifications as $notification ) {
+			if ( $this->show_notification( $notification ) ) {
+				echo $notification;
+			}
+		}
+
+		// Clear the local stored notifications.
+		$this->clear_notifications();
+	}
 
-		foreach ( $this->notifications as $key => $notification ) {
+	/**
+	 * Return the notifications sorted on type and priority
+	 *
+	 * @return array|Yoast_Notification[] Sorted Notifications
+	 */
+	public function get_sorted_notifications() {
+		$notifications = $this->notifications;
+		if ( empty( $notifications ) ) {
+			return array();
+		}
 
-			$id = $notification->get_id();
+		// Sort by severity, error first.
+		usort( $notifications, array( $this, 'sort_notifications' ) );
 
-			if ( in_array( $id, $ids ) ) {
-				unset( $this->notifications[ $key ] );
-				continue;
+		return $notifications;
+	}
+
+	/**
+	 * Sort on type then priority
+	 *
+	 * @param Yoast_Notification $a Compare with B.
+	 * @param Yoast_Notification $b Compare with A.
+	 *
+	 * @return int 1, 0 or -1 for sorting offset.
+	 */
+	private function sort_notifications( Yoast_Notification $a, Yoast_Notification $b ) {
+		$a_type = $a->get_type();
+		$b_type = $b->get_type();
+
+		if ( $a_type === $b_type ) {
+			$a_priority = $a->get_priority();
+			$b_priority = $b->get_priority();
+
+			if ( $a_priority === $b_priority ) {
+				return 0;
 			}
 
-			$ids[] = $id;
+			return ( ( $a_priority < $b_priority ) ? 1 : -1 );
```

Comment:
I would not call that simple or at least not easy to read.
Simplified by using `bccmp`.